File: src/main/java/land/leets/domain/user/domain/repository/UserRepository.java
package land.leets.domain.user.domain.repository;

import land.leets.domain.user.domain.User;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;
import java.util.UUID;

public interface UserRepository extends JpaRepository<User, UUID> {
    Optional<User> findBySub(String sub);
}

File: src/main/kotlin/land/leets/domain/user/domain/repository/UserRepository.kt
package land.leets.domain.user.domain.repository

import land.leets.domain.user.domain.User
import org.springframework.data.jpa.repository.JpaRepository
import java.util.UUID

interface UserRepository : JpaRepository<User, UUID> {
    fun findBySub(sub: String): User?
    fun findByEmail(email: String): User?
}
